### PR TITLE
sphinx.ext.extlinks: Add `extlink-{name}` CSS class to links

### LINKED ISSUE #11745: sphinx.ext.extlinks: option for literal formatting?
**Is your feature request related to a problem? Please describe.**

In the Python devguide, we have an extension to add custom roles that add links.

https://github.com/python/devguide/blob/2e9e657b53830806d202df70cbed3353afcdb5d1/_extensions/custom_roles.py

For example, ``` :cpy-file:`.github/CODEOWNERS` ``` will expand to `https://github.com/python/cpython/blob/main/.github/CODEOWNERS`:

image

https://devguide.python.org/core-developers/experts/

This could of course be replaced with `sphinx.ext.extlinks` with this in `conf.py`:

```python
_repo = "https://github.com/python/cpython"
extlinks = {
    "cpy-file": (f"{_repo}/blob/main/%s", "%s"),
    "gh-label": (f"{_repo}/labels/%s", "%s"),
}
```

However, the custom extension output as literal text, and `extlinks` does not apply any formatting and looks like this:

image

**Describe the solution you'd like**

Add some sort of option to use literal formatting for `extlinks`.

This sort of thing adds literal formatting. But it would need some sort of option to enable it:

```diff
diff --git a/sphinx/ext/extlinks.py b/sphinx/ext/extlinks.py
index 173df4dde..af79f716d 100644
--- a/sphinx/ext/extlinks.py
+++ b/sphinx/ext/extlinks.py
@@ -106,7 +106,7 @@ def make_link_role(name: str, base_url: str, caption: str) -> RoleFunction:
                 title = full_url
             else:
                 title = caption % part
-        pnode = nodes.reference(title, title, internal=False, refuri=full_url)
+        pnode = nodes.reference(title, '', nodes.literal(rawtext, text), internal=False, refuri=full_url)
         return [pnode], []
     return role
```

**Describe alternatives you've considered**

Keep the extension and custom roles. 

Use CSS?

### PR DESCRIPTION
**Pull request by hugovk**
### Feature or Bugfix

- Feature

### Purpose
- Allow theme authors and docs builders flexibility to apply whatever custom formatting to an extlink without the need for custom extensions.

### Relates
- Fixes https://github.com/sphinx-doc/sphinx/issues/11745.

### CODE REVIEW

File: sphinx/ext/extlinks.py
Comment (picnixz) [inline sphinx/ext/extlinks.py]: Technically speaking, the `name` will be used as a role name. In particular, it must be a simple name, namely it matches `'(?:(?!_)\w)+(?:[-._+:](?:(?!_)\w)+)*`. Therefore, I think you don't need to escape the name (and you will have a patch of exactly one line  + two lines of test, an improvement of 25%!)

Comment (hugovk) [reply sphinx/ext/extlinks.py]: Nice, I've removed the `escape` 🚀

**Comment by picnixz** [comment]
I personally am good with that approach. @AA-Turner I'll be out from now on, so I'll let you merge this one if you want to add some comments or change the class name.

**Review by chrisjsewell — APPROVED** [review]
all good cheers

**Pull request merged**
